docs/benchmarking.md

```diff
@@ -4,28 +4,137 @@ toc: true
+
+Given the public key $$\textsf{pk} = (\tilde{g}, \tilde{X}, \tilde{Y}_1, \dots, \tilde{Y}_{r+1})$$, the message vector $$\textbf{m} = (m_1, \dots, m_r)$$, and the signature $$\sigma = (m', \sigma_1, \sigma_2)$$, the verification algorithm works as follows:
+Check whether $$\sigma_1$$ is the multiplicative identity of $$\mathbb{G}_1$$ and output $$0$$ if yes.
+Lastly, output $$1$$ if $$e(\sigma_1, \tilde{X} \cdot \prod_{j = 1}^r{\tilde{Y}_j^{m_j} \cdot \tilde{Y}_{r+1}^{m'}}) = e(\sigma_2, \tilde{g})$$ holds, and $$0$$ if not.
+Here $$e$$ denotes the pairing operation.
+
+We test the verification operation using messages of length one and of length 10.
+```java
+@State(Scope.Thread)
+public class PS18VerifyBenchmark {
+	
+    // Test with one message and ten
+    @Param({"1", "10"})
+    int numMessages;
+	
+    PS18SignatureScheme scheme;
+    PlainText plainText;
+    Representation signatureRepr;
+    Representation verifyKeyRepr;
+	
+    // The setup method that creates the signature and verification key 
+    //  used by the verify benchmark
+    @Setup(Level.Iteration)
+    public void setup() {
+        PSPublicParameters pp = new PSPublicParameters(new MclBilinearGroup());
+        scheme = new PS18SignatureScheme(pp);
+        SignatureKeyPair<? extends PS18VerificationKey, ? extends PS18SigningKey> keyPair =
+                scheme.generateKeyPair(numMessages);
+        RingElementPlainText[] messages = new RingElementPlainText[numMessages];
+        for (int i = 0; i < messages.length; i++) {
+            messages[i] = new RingElementPlainText(pp.getZp().getUniformlyRandomElement());
+        }
+        plainText = new MessageBlock(messages);
+        signature = scheme.sign(plainText, keyPair.getSigningKey());
+        verificationKey = keyPair.getVerificationKey();
+        // Computations in sign and/or key gen may be done non-blocking.
+        // To make sure these are not done as part of the verification benchmark, 
+        //  we force the remaining computations to be done blocking via getRepresentation()
+        signatureRepr = signature.getRepresentation();
+        verifyKeyRepr = verificationKey.getRepresentation();
+    }
+	
+    // The benchmark method. Includes settings for JMH
+    @Benchmark
+    @BenchmarkMode(Mode.SingleShotTime)
+    @Warmup(iterations = 3, batchSize = 1)
+    @Measurement(iterations = 10, batchSize = 1)
+    @OutputTimeUnit(TimeUnit.MILLISECONDS)
+    public Boolean measureVerify() {
+        // Running the signing or key gen algorithms may have done precomputations
+        //  for the verification key and/or signature.
+        // To reset these precomputation such that they do not make the verification
+        //  algorithm faster than it would be without, we recreate the objects
+        //  without precomputations.
+        signature = scheme.restoreSignature(signatureRepr);
+        verificationKey = scheme.restoreVerificationKey(verifyKeyRepr);
```

Comment:
This could actually be discussed. This highly depends on what you want to benchmark. If you want to verify multiple signatures on the same key that you keep in memory, you should just use the same verificationKey object every time.

The current benchmark is "I get a list of unknown signatures/verification keys and need to verify them all".